### ISSUE TITLE
Importer problem + solutions

### ISSUE THREAD

**Issue by nomansbeauty** [Webcreations907/WBC-Importer-extension#45]
Hi,

Thanks for making this great extension. We are using it since an year now. We often face problems (or our customers) due to which support queue gets long. Most customers panic due to not understanding of server requirements.

<strong>Problems that can be solved:</strong>
1. No incremental loading of data - asynchronous data loading will bypass server php execution time limits, which is often 30 sec on shared hosting (godaddy, hostgator, bluehost, etc..).
2. Should function properly with PHP Memory Limit = 64mb or more
3. There are plugins bypassing these limits efficiently such as All-in-One WP Migration:
   https://wordpress.org/plugins/all-in-one-wp-migration/
4. Should give proper notifications or warnings in case of import demo is stopped or broken or network connection is lost or host server memory consumed or originating server is not responding due to lack of resources, or successful without problems etc... So user knows what caused the stoppage of import demo content.
5. And if demo content is stopped in mid for any reason, it should erase whatever was imported. So re-importing content wont show duplicated content issue (such as menu items).
6. Plugin should detect server settings and display user warnings that his server lacks certain configurations that he should make before starting import process.

This will save tons of headache and issues for theme authors and support teams. Please let me know if you need any help (I can help in testing and making some donation for it too).
Thank you

**Comment by Webcreations907** [comment]
Hello,

When creating/modifying this for a theme for Themeforest I've been working on, I had it in mind that it would be importing small sized placeholder images as most themes on Themeforest don't include the preview/demo images. 

I've thought about background loading of demo data, just haven't had the time to actually start putting it together as I'm trying to get my theme done as well. 

I know that it would be creating a custom import process as right now it just uses the WP_importer, the duplicate menus it caused by the WP_importer which has been a issue for some time as there are several posts on the web for that. (You could also filter the import process to check for duplicate menu items so that doesn't happen.)

Will look over the info you provided, and see what the best route would be to go.

You're more than welcome to fork it and change it up how you'd like though in the mean time.

**Comment by contempoinc** [comment]
I'd be willing to throw some money into making this happen, 95% of the support requests I get for my RE7 theme are buyers not able to import the data due to server limits, etc…

**Comment by contempoinc** [comment]
Would changing set_time_limit(0) to ini_set('max_execution_time', 0) make any difference?

**Comment by Webcreations907** [comment]
@contempoinc should be the same effect, ini_set may be disabled(safe mode,etc) on some hosts though.

**Comment by Webcreations907** [comment]
@nomansbeauty I've checked out that plugin briefly on the process used, definitely wouldn't be able to use the same process or we'd end up with 100gb themes when including multiple demo options, which would cause another issue. :) 

I will try to work on this when I get some time, have had a plan for it for awhile just been really busy.

**Comment by contempoinc** [comment]
@Webcreations907 @nomansbeauty  I'd be willing to pay to get this done, it would save me hours and hours of support.

**Comment by nomansbeauty** [comment]
@contempoinc Thanks for offer. Lets talk about it sometime this week, I have solid offshore team of developers who can improve this, may cost a bit high, but we can together handle the cost I believe. Please connect with me on skype ID: nomans_beauty

Thanks alot @Webcreations907 for details, I will be looking forward to improvements. Do you guys know any other paid plugin for this? I searched but couldn't find anything like this.

Thanks

**Comment by contempoinc** [comment]
Shot ya an add on Skype

Cheers,

Chris Robinson
www.contempoinc.com
twitter.com/contempoinc

**Comment by Webcreations907** [comment]
@nomansbeauty no problem, I don't know of any plugins, pretty sure there isn't any other for Redux either unless @dovy is working on one?

**Comment by dovy** [comment]
@kprovance is said to be.  ;)

**Comment by erropix** [comment]
Dear @contempoinc,
I got the RE7 theme, it's there's lot of issues in the demo content, half of the images return 404 error and the widget.json is not relevant to your theme, it just contains "Widget WBC Demo Importer" for a sidbar that do not exists in the theme.

aside from that,
I work frequently on theme installations and I know how the process should be in perfect world.
I am currently working on a complete export/import solution that will solve all the headache of support and server issues. it wont be using WP_Importer and wont be like site movers.
If any one have ideas to commit please let me know

**Comment by nomansbeauty** [comment]
@ERROPiX  glad to hear that. Good luck and do share with us once ready :)

**Comment by kanawanzx** [comment]
about time limit. if internet in my country so slow then and the import demo is not apply. it to affect quality of theme 
i think . we need careful with internet in Least Developed Countries.
this is my idea.
sorry about my english not good

**Comment by Moutaz-homsi** [comment]
any news related to this issue guys ?
if someone start to fix it maybe i can help .

**Comment by Moutaz-homsi** [comment]
Hi guys 
so how do you solve this issue for your themes ?
i found this code https://github.com/proteusthemes/one-click-demo-import , can we implement the ajaxing functions that they use it to import files here ?
i prefer to use this code because i use reduxframework and its easier to import demo data from theme option rather than from a 3d party plugin or from another admin page
let me know what happen , thanks :-)

**Comment by kprovance** [comment]
I ended up writing my own from scratch, that hooks into the WP importer and uses ajax to cycle through various import files (and makes a nice progress bar).  It doesn't suport rollback tho.  That's way too complicated considering the interaction with the database.

**Comment by nomansbeauty** [comment]
@kprovance  :+1:   Would be great if you can share the link with everyone here. Thank you

**Comment by kprovance** [comment]
There is no link.  :)

I spent weeks getting that complex piece of code to work properly.  The _last_ think I want to do is just give it away and have to answer 100 questions about how to get it to work.  I already do that with Redux for free (mostly to a bunch of people who are too cheap to donate or even say 'thank you' half the time..  Oh, and it's a Redux extension, not a one size fits all solution.  :)

If I ever release it, it's been via the Redux website as a premium extension, or Code Canyon.

**Comment by nomansbeauty** [comment]
> If I ever release it, it's been via the Redux website as a premium extension, or Code Canyon.

Sounds good.

**Comment by Moutaz-homsi** [comment]
> If I ever release it, it's been via the Redux website as a premium extension, or Code Canyon.

Sounds great, this will benefit both of us .

**Comment by erropix** [comment]
Hi guys,
I want to start project to solve the export/import problems of WordPress but I didn't found some to help.
If anyone want to contribute to this project please contact me at admin@erropix.com
Thanks

**Comment by Moutaz-homsi** [comment]
Hi @ERROPiX , i would love to , but because mr kprovance has done the same task i think we can buy it from him as redux premium extension, codecanyon plugin or anything mr kprovance prefer .
i do not think redo the code again would be good for us while its already there .
@kprovance  please keep us updated if you would release it for sell soon :-)

**Comment by QuinnWilton** [comment]
❤️ 💚 💙 💛 💜
